src/modules/job-exec/job-exec.c

```diff
@@ -1094,10 +1094,170 @@ static int configure_implementations (flux_t *h, int argc, char **argv)
     return 0;
 }
 
-static int unload_implementations (void)
+static flux_future_t *lookup_namespace_roots (struct job_exec_ctx *ctx)
+{
+    struct jobinfo *job = zhashx_first (ctx->jobs);
+    flux_future_t *fall = NULL;
+    flux_future_t *f = NULL;
+
+    while (job) {
+        if (job->running) {
+            if (!fall) {
+                if (!(fall = flux_future_wait_all_create ())) {
+                    flux_log_error (ctx->h, "flux_future_wait_all_create");
```

Comment:
Error messages could be improved in this function.

Reply:
I meant to say "this commit" rather than "this function".  The error messages all look like they not contain enough context to really run anything down.